### PR TITLE
feat(javascript): add `getSecuredApiKey` and validity check

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1773

### Changes included:

- [x] add `getSecuredApiKey` and `getSecuredApiKeyRemainingValidity` helpers
- [x] update deprecated method
- [x] fix wrong descriptions

### CODE REVIEW

File: templates/javascript/clients/client/model/clientMethodProps.mustache [hunk 1]
```diff
@@ -95,6 +95,47 @@ export type WaitForApiKeyOptions = WaitForOptions & {
...
+}
+
+export type GetSecuredApiKeyRemainingValidityOptions = {
+  /**
+   * The secured API key generated with the `generateSecuredApiKey` method.
+   */
+  securedApiKey: string;
+}
+
+export type SecuredApiKeyRestrictions = SearchParamsObject & {
```
Comment (Fluf22) [inline templates/javascript/clients/client/model/clientMethodProps.mustache:118]: `SearchParamsObject` should be optional, no?

**Review by Fluf22 — DISMISSED** [review]
Just a qq

File: templates/javascript/clients/client/model/clientMethodProps.mustache (hunk 1 above)
Comment (shortcuts) [reply templates/javascript/clients/client/model/clientMethodProps.mustache:118]: I've also moved it to its own field like in the other clients

**Review by Fluf22 — DISMISSED** [review]
perfect.gif

**Review by Fluf22 — APPROVED** [review]
fr fr

**Pull request merged**